sktree/stats/utils.py

```diff
@@ -0,0 +1,243 @@
+from typing import Tuple
+
+import numpy as np
+from numpy.typing import ArrayLike
+from scipy.stats import entropy
+from sklearn.metrics import (
+    balanced_accuracy_score,
+    mean_absolute_error,
+    mean_squared_error,
+    roc_auc_score,
+)
+from sklearn.utils.validation import check_X_y
+
+from sktree._lib.sklearn.ensemble._forest import ForestClassifier
+from sktree._lib.sklearn.tree import DecisionTreeClassifier
+
+
+def _mutual_information(y_true, y_pred):
+    H_YX = np.mean(entropy(y_pred, base=np.exp(1)))
+    _, counts = np.unique(y_true, return_counts=True)
+    H_Y = entropy(counts, base=np.exp(1))
+    return max(H_Y - H_YX, 0)
```

Comment:
This is wrong. The method in `_might.py` is buggy. The correct one is in `hyppo`.

Reply:
I'll replace. I don't use this method currently.

Reply:
The only thing is `axis=1` within the entropy computation for H_YX right?

Reply:
As you already implemented the multi-class, this should fix it.